daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -143,6 +150,19 @@ object SValue {
 
   final case class SMap(value: HashMap[String, SValue]) extends SValue
 
+  final case class SGenMap(insertion: Int, value: HashMap[SGenMap.Key, (Int, SValue)])
+      extends SValue
+
+  object SGenMap {
+    case class Key(v: SValue) {
```

Comment:
Just to check, this will throw an exception if the key is not valid, right? e.g. if it contains a function.

Reply:
I made that more explicit